docs/src/sections/MenuItemSection.js

```diff
@@ -22,7 +22,7 @@ export default function MenuItemSection() {
         <li><code>divider</code>: Adds an horizontal divider between sections</li>
         <li><code>disabled</code>: shows the item as disabled, and prevents the onclick</li>
         <li><code>eventKey</code>: passed to the callback</li>
-        <li><code>onSelect</code>: a callback that is called when the user clicks the item.</li>
+        <li><code>onSelect</code>: a callback that is called when the user clicks the item. <b>NOTE:</b> the containing DropdownMenu actually re-writes this property on its direct child elements. For simple (direct) uses of MenuItem, this will not affect you. However, if you wrap a MenuItem in your own component, be careful! You <em>must</em> handle the onSelect property accordingly, or the <a href="https://github.com/react-bootstrap/react-bootstrap/issues/2429">dropdown will not close</a> after a user selects a menu item!</li>
```

Comment:
>  Phrased another way, the DropdownMenu expects a component that implements the same API as a MenuItem.

Okay, I see now where you're coming from. I'll be stubborn and still say it's suprising, and would be nice to handle either internally (perhaps configurable via an API on the menu) or by turning Select into an actual custom event that bubbles (not sure if React supports that yet) — but I agree this does make more sense when you put it in terms of essentially: a dropdown menu only really supports having MenuItems as its direct children. By wrapping them I ventured into somewhat uncharted territory, and so I had to be aware of some internal details.

Thanks for your patience :-)